src/pubsub/ua_pubsub_reader.c

```diff
@@ -1060,9 +1136,39 @@ UA_DataSetReaderConfig_copy(const UA_DataSetReaderConfig *src,
     return UA_STATUSCODE_GOOD;
 }
 
+void
+UA_DataSetReaderConfig_clear(UA_DataSetReaderConfig *cfg) {
+    UA_String_deleteMembers(&cfg->name);
+    UA_Variant_deleteMembers(&cfg->publisherId);
+    UA_DataSetMetaDataType_deleteMembers(&cfg->dataSetMetaData);
+    UA_ExtensionObject_clear(&cfg->messageSettings);
+    UA_ExtensionObject_clear(&cfg->transportSettings);
+    if (cfg->subscribedDataSetType == UA_PUBSUB_SDS_TARGET) {
+        UA_TargetVariables_clear(&cfg->subscribedDataSet.subscribedDataSetTarget);
+    }
+
+}
+
+UA_StatusCode
+UA_Server_DataSetReader_getState(UA_Server *server, UA_NodeId dataSetReaderIdentifier,
+                               UA_PubSubState *state) {
+
+    if(!state)
+        return UA_STATUSCODE_BADINVALIDARGUMENT;
+
+    UA_DataSetReader *currentDataSetReader =
+        UA_ReaderGroup_findDSRbyId(server, dataSetReaderIdentifier);
+    if(!currentDataSetReader)
+        return UA_STATUSCODE_BADNOTFOUND;
+
+    *state = currentDataSetReader->state;
+    return UA_STATUSCODE_GOOD;
+}
+
```

Comment:
Some functions can be formatted a little bit more compact.

Reply:
ok, changed it here: https://github.com/open62541/open62541/pull/3899